docs: Fix issues found in Queeny's test (#507)
pingcap/tidb-operator: docs/local-dind-tutorial.md

    > **Note:** [Legacy Docker Toolbox](https://docs.docker.com/toolbox/toolbox_install_mac/) users must migrate to [Docker for Mac](https://store.docker.com/editions/community/docker-ce-desktop-mac) by uninstalling Legacy Docker Toolbox and installing Docker for Mac, because DinD cannot run on Docker Toolbox and Docker Machine.

>     > **Note:** `kubeadm` validates installed Docker version during the installation process. If you are using Docker later than 18.06, there would be warning messages. The cluster might still be working, but it is recommended to use a Docker version between 17.03 and 18.06 for better compatibility. You can find older versions of docker at [here](https://download.docker.com/).

Reviewer: ```suggestion
    > **Note:** `kubeadm` validates installed Docker version during the installation process. If you are using Docker later than 18.06, there would be warning messages. The cluster might still be working, but it is recommended to use a Docker version between 17.03 and 18.06 for better compatibility. You can find older versions of docker [here](https://download.docker.com/).
```

    
    You can check if your filesystem supports `d_type` using command `xfs_info / | grep ftype`, where `/` is the data directory of you installed Docker daemon.

    If your root directory `/` uses XFS without `d_type` support, but there is another partition does, or is using another filesystem, it is also possible to change the data directory of Docker to use that partition.

    Assume a supported filesystem is mounted at path `/data`, use the following instructions to let Docker use it:

    ```sh
    # Create a new directory for docker data storage
    mkdir -p /data/docker

    # Stop docker daemon
    systemctl stop docker.service

    # Make sure the systemd directory exist
    mkdir -p /etc/systemd/system/docker.service.d/

    # Overrite config
    cat << EOF > /etc/systemd/system/docker.service.d/docker-storage.conf
    [Service]
    ExecStart= 
    ExecStart=/usr/bin/dockerd --data-root /data/docker -H fd:// --containerd=/run/containerd/containerd.sock
    EOF

    # Restart docker daemon
    systemctl daemon-reload
    systemctl start docker.service
    ```

## Step 1: Deploy a Kubernetes cluster using DinD

> Install and set up a Kubernetes cluster (version 1.12) using DinD for TiDB Operator with the script in our repository.

Reviewer: ```suggestion
Install and set up a Kubernetes cluster (version 1.12) using DinD for TiDB Operator with the script in our repository:
```

$ kubectl get nodes -o wide
NAME          STATUS   ROLES    AGE     VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE                       KERNEL-VERSION               CONTAINER-RUNTIME
kube-master   Ready    master   11m     v1.12.5   10.192.0.2    <none>        Debian GNU/Linux 9 (stretch)   3.10.0-957.12.1.el7.x86_64   docker://18.9.0
kube-node-1   Ready    <none>   9m32s   v1.12.5   10.192.0.3    <none>        Debian GNU/Linux 9 (stretch)   3.10.0-957.12.1.el7.x86_64   docker://18.9.0
kube-node-2   Ready    <none>   9m32s   v1.12.5   10.192.0.4    <none>        Debian GNU/Linux 9 (stretch)   3.10.0-957.12.1.el7.x86_64   docker://18.9.0
kube-node-3   Ready    <none>   9m32s   v1.12.5   10.192.0.5    <none>        Debian GNU/Linux 9 (stretch)   3.10.0-957.12.1.el7.x86_64   docker://18.9.0
```

## Step 2: Install TiDB Operator in the DinD Kubernetes cluster

Once the k8s cluster is up and running, we can install TiDB Operator into it using `helm`:

```sh
$ helm install charts/tidb-operator --name=tidb-operator --namespace=tidb-admin --set scheduler.kubeSchedulerImageName=mirantis/hypokube --set scheduler.kubeSchedulerImageTag=final
```

> Then wait few minutes until operator is running:

Reviewer: ```suggestion
Then wait a few minutes until TiDB Operator is running:
```

- `root` access or permissions to operate with the Docker daemon.

- Supported filesystem

    If the host machine uses an XFS filesystem (default in CentOS 7), it must be formatted with `ftype=1` to enable `d_type` support, see [Docker's documentation](https://docs.docker.com/storage/storagedriver/overlayfs-driver/) for details.
    

>     You can check if your filesystem supports `d_type` using commaind `xfs_info / | grep ftype`, where `/` is the data directory of you installed Docker daemon.

Reviewer: ```suggestion
    You can check if your filesystem supports `d_type` using command `xfs_info / | grep ftype`, where `/` is the data directory of you installed Docker daemon.
```

    If your root directory `/` uses XFS without `d_type` support, but there is another partition does, or is using another filesystem, it is also possible to change the data directory of Docker to use that partition.

    Assume a supported filesystem is mounted at path `/data`, use the following instructions to let Docker use it:

    ```sh
    # Create a new directory for docker data storage
    mkdir -p /data/docker

    # Stop docker daemon
    systemctl stop docker.service

    # Make sure the systemd directory exist
    mkdir -p /etc/systemd/system/docker.service.d/

    # Overrite config
    cat << EOF > /etc/systemd/system/docker.service.d/docker-storage.conf
    [Service]
    ExecStart= 

>     ExecStart=/usr/bin/dockerd -g /data/docker -H fd:// --containerd=/run/containerd/containerd.sock

Reviewer 2: As from docker-ce 17.05.0, the `-g` or `--graph` flag has been replaced by `--data-root` flag. refer [here](https://docs.docker.com/engine/deprecated/)

## Step 3: Deploy a TiDB cluster in the DinD Kubernetes cluster

By using `helm` along with TiDB Operator, we can easily set up a TiDB cluster:

```sh
$ helm install charts/tidb-cluster --name=demo --namespace=tidb
```

> And wait a few minutes for all TiDB components get created and ready:

Reviewer: ```suggestion
And wait a few minutes for all TiDB components to get created and ready:
```

```sh

> # Use Ctrl + C to exit watch mode

Reviewer: ```suggestion
# Use `Ctrl + C` to exit watch mode
```

- Access TiDB using the MySQL client

    Before you start testing your TiDB cluster, make sure you have installed a MySQL client.

    1. Use `kubectl` to forward the host machine port to the TiDB service port:

        ```sh
        $ kubectl port-forward svc/demo-tidb 4000:4000 --namespace=tidb
        ```

    2. Then, to connect to TiDB using the MySQL client, open a new terminal tab or window and run the following command:

        ```sh
        $ mysql -h 127.0.0.1 -P 4000 -u root
        ```

- View the monitor dashboards

    1. Use `kubectl` to forward the host machine port to the Grafana service port:

        ```sh
        $ kubectl port-forward svc/demo-grafana 3000:3000 --namespace=tidb
        ```

        If the proxy is set up sucessfully, it will print something like `Forwarding from 0.0.0.0:3000 -> 3000`, press `Ctrl + C` to stop the proxy and exit.

>     2. Open your web browser at http://localhost:3000 to access the Grafana monitoring interface.

Reviewer: ```suggestion
    2. Then, open your web browser at http://localhost:3000 to access the Grafana monitoring interface.
```

You can scale out or scale in the TiDB cluster simply by modifying the number of `replicas`.

> 1. Edit the `charts/tidb-cluster/values.yaml` file with your preffered text editor.

Reviewer: ```suggestion
1. Edit the `charts/tidb-cluster/values.yaml` file with your preferred text editor.
```

> **Note:** If you need to scale in TiKV, the consumed time depends on the volume of your existing data, because the data needs to be migrated safely.

Use `kubectl get pod -n tidb` to verify the number of each compoments equal to values in the `charts/tidb-cluster/values.yaml` file, and all pods are in `Running` state.

## Upgrade the TiDB cluster

> 1. Edit the `charts/tidb-cluster/values.yaml` file with your preffered text editor.

Reviewer: ```suggestion
1. Edit the `charts/tidb-cluster/values.yaml` file with your preferred text editor.
```

    ```sh
    $ manifests/local-dind/dind-cluster-v1.12.sh stop
    ```

>     You can use `docker ps` to verify there are no docker container running.

Reviewer: ```suggestion
    You can use `docker ps` to verify that there is no docker container running.
```
